### PR TITLE
Update za.private.json

### PR DESCRIPTION
**Pull request by Humairaa127**
Update South Africa public holidays for 2022

### CODE REVIEW

File: assets/operatingInfo/za.private.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-    "04/05/2021":"Family Day",
-    "04/27/2021":"Freedom Day",
-    "05/01/2021":"Workers Day",
-    "06/16/2021":"Youth Day",
-    "08/09/2021":"National Womens Day",
-    "09/24/2021":"Heritage Day",
-    "12/16/2021":"Day of Reconciliation",
-    "12/25/2021":"Christmas",
-    "12/27/2021":"Boxing Day"
+    "01/02/2022":"New Years Day",
```
Comment (murilovmachado) [inline assets/operatingInfo/za.private.json:4]: Is the update New Year's day really Jan 2nd?

Comment (Humairaa127) [reply assets/operatingInfo/za.private.json]: > Is the updated New Year's day really Jan 2nd?

Thanks, yes for 2023.

**Review by murilovmachado — APPROVED** [review]

**Comment by GPaoloni** [comment]
Looking the changes required here, I realize this is not ergonomic in the sense that it must be changed every year. We could improve the code to only consider day and month. Is that something desired @dee-luo @Humairaa127?

**Comment by stephenhand** [comment]
> Looking the changes required here, I realize this is not ergonomic in the sense that it must be changed every year. We could improve the code to only consider day and month. Is that something desired @dee-luo @Humairaa127?

I'm guessing they need to be updated every year anyway, since public holiday day dates change every year (Easter, when the day of a regular holiday falls on the weekend they move the public holiday to the following Monday etc.)

I think the only way we could take the legwork out of this would be to import them from an external data source

**Comment by Humairaa127** [comment]
> Looking the changes required here, I realize this is not ergonomic in the sense that it must be changed every year. We could improve the code to only consider day and month. Is that something desired @dee-luo @Humairaa127?

Yes, this is on the list of possible improvements i'm looking at for the year but not right now since it's a low effort process in it's current state, so we have time to work on it. As Steve mentioned, many dates change over years, and there are also certain countries who's holidays work on different calendars, not Gregorian. I've begun some basic research into libraries that are available for this and begin a team discussion soon to brainstorm solutions and assess viability.

**Pull request merged**